plot-file can now plot column 1 against remaining columns in data file.

lgrozinger opened the pull request:
plot-file takes &optional argument which should be T if the first
column in the data-file contains x-values, and omitted (or NIL)
otherwise.

The new DEFUNs get-cmd-string and get-next-cmd-string return the
appropriate gnuplot command strings depending on the value of
x-in-file argument.

lgrozinger committed:
plot-file can now plot column 1 against remaining columns in data file.

plot-file takes &optional argument which should be T if the first
column in the data-file contains x-values, and omitted (or NIL)
otherwise.

The new DEFUNs get-cmd-string and get-next-cmd-string return the
appropriate gnuplot command strings depending on the value of
x-in-file argument.

volkers commented:
Hello Lewis,

thanks for contributing!

I like your change but some things before I would merge it:

- for me, both GET-CMD-STRING and GET-NEXT-CMD-STRING are rather
  local functions. I don't see any use outside of PLOT-FILE.
  Therefore I would prefer having them defined locally using FLET.

- Please update also the documentation string of PLOT-FILE.


An idea for further improvement:

Would it be possible to change:

   `(defun plot-file (data-file &optional x-in-file)`
to

  `(defun plot-file (data-file &key (x))`

meaning something like:

"... optional:
   :x nr    number of column used as x (default using index)"

so that it could also be another column used as x?
Thoughts?

Regards
Volker

lgrozinger commented:
Hello Volker,

Firstly, I really like vgplot and have found it very useful, thank you for providing it.

I agree with all of your suggestions, I'm now working on making these changes. Thank you for the feedback! 

BW
Lewis

lgrozinger committed:
PLOT-FILE takes &key parameter X-COL, the column to use as x values.

Documentation string of PLOT-FILE updated with new parameter.

GET-CMD-STRING and GET-NEXT-CMD-STRING combined into one DEFUN and
defined using FLET in PLOT-FILE.

volkers commented:
Hello Lewis,
your code is merged and published.
Thanks and have a nice day!
Volker

volkers commented:
Hello Lewis,
would you like to be mentioned as a contributor in the source code of vgplot.lisp?
/Volker

lgrozinger commented:
Hi Volker,


I have no objections if you think it is appropriate.


BW

Lewis

________________________________
From: Volker Sarodnick <notifications@github.com>
Sent: 06 November 2018 08:03:32
To: volkers/vgplot
Cc: Lewis Grozinger (Student); Author
Subject: Re: [volkers/vgplot] plot-file can now plot column 1 against remaining columns in data file. (#4)


Hello Lewis,
would you like to be mentioned as a contributor in the source code of vgplot.lisp?
/Volker

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/volkers/vgplot/pull/4#issuecomment-436163709>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AXdtDZlIpMO31AHmL2G8_6iUvLEVkYelks5usUJUgaJpZM4X_jgN>.

volkers commented:
Ok, done.

/Volker
